lib/cisco_node_utils/domain_name.rb

```diff
@@ -0,0 +1,54 @@
+# Domain Name provider class
+#
+# October 2015, Bryan Jen
+#
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require File.join(File.dirname(__FILE__), 'node_util')
+
+module Cisco
+  # DomainName- node utility class for domain name configuration
+  class DomainName < NodeUtil
+    attr_reader :name
+
+    def initialize(name, instantiate=true)
```

Comment:
I am looking into this for #23; when looking at the README_YAML.md spec on how `config_get` and `config_get_token` keys work, there doesn't seem to be a way to extract both non-vrf and vrf values. It's either/or, and the vrf must be specified ahead of time. Thus we would need to gather all the non-vrf domains, then get a list of vrfs and loop over them to get all of the vrf domains. There is no node-utils api for getting the vrfs yet, so that would have to be written as a prerequisite.